site-src/geps/gep-718.md

```diff
@@ -0,0 +1,223 @@
+introduce `Service` as the default `kind` for `RouteForwardTo.BackendRef` and
+`HTTPRouteForwardTo.BackendRef`.
+
+## Motivation
+
+Within `RouteForwardTo` (and `HTTPRouteForwardTo` by extension) there exists
+two mechanisms to specify the upstream network endpoint where the traffic should
+be forwarded to:
+- `ServiceName`: the name of the Kubernetes Service
+- `BackendRef`: a LocalObjectReference to a resource, this is an extension point in the API
+
+While working on [#685](https://github.com/kubernetes-sigs/gateway-api/pull/685),
+it came to light that:
+1. `BackendRef` itself could be used to point to a Kubernetes Service
+2. With [GEP-709](https://github.com/kubernetes-sigs/gateway-api/pull/711),
+there will be a need to introduce a new `namespace` field which will enable use-cases to forward
+traffic to a services and backends located in different namespaces
+
+While (1) can be fixed with more validation and documentation, it only solves for
+the specific case. (2) requires adding two fields: `serviceNamespace`
+and `BackendRef.Namespace` - something we would like to avoid since the `serviceName`
+field was introduced only as a shortcut and adding more service-specific fields
+defeats the original purpose.
+
+These problems are in addition to the original problem that
+[#685](https://github.com/kubernetes-sigs/gateway-api/pull/685) attempts to solve:
+clarifying port requirements when a port is required or not.
+We have been updating documentation to tackle such corner-cases but that
+continues to result in more and more elaborations. Some excerpts from our
+existing documentation:
+```
+	// ServiceName refers to the name of the Service to mirror matched requests
+	// to. When specified, this takes the place of BackendRef. If both
+	// BackendRef and ServiceName are specified, ServiceName will be given
+	// precedence.
+
+	// BackendRef is a local object reference to mirror matched requests to. If
+	// both BackendRef and ServiceName are specified, ServiceName will be given
+	// precedence.
+
+	// Weight specifies the proportion of HTTP requests forwarded to the backend
+	// referenced by the ServiceName or BackendRef field. This is computed as
+
+	// Port specifies the destination port number to use for the
+	// backend referenced by the ServiceName or BackendRef field.
+	// If unspecified, the destination port in the request is used
+	// when forwarding to a backendRef or serviceName.
+```
+
+
+## Proposal
+
+- Remove the `ServiceName` field from `RouteForwardTo` and `HTTPRouteForwardTo`
+- Introduce `Service` as a default for `BackendRef.Kind`
+- Apply the same to HTTPRequestMirrorFilter to avoid issues like [#655](https://github.com/kubernetes-sigs/gateway-api/issues/655)
+
+## API
+
+
+The updated `RouteForwardTo` will be:
+(`HTTPRouteForwardTo` has been omitted for brevity)
+
+```go
+// RouteForwardTo defines how a Route should forward a request.
+type RouteForwardTo struct {
```

Comment:
Is it possible to put an indicator as to if there are changes to the long comment text

like this?

```
// no change
// blah blah blah    <<
// no change
```

Reply:
@bowei I added another summarized diff in the proposal for a clearer understanding of what is changing. PTAL.